### ISSUE TITLE
NginxUnit isn't be reconfigured and app responds with 408

### ISSUE THREAD

**Issue by vsevolod-ch** [nginx/unit#222]
Hello!

We have some problems with NginxUnit.

Some days ago I found that our server was answering very long. Responses have been being received only after 2 or 3 seconds waiting for some time. But some time later the time of response was restored to 100 or 150 ms. But it has been being repeated several times. There was no problem either memory or  cpu or disk space or database access...

Today our server is down. Server is responding with 408 «Request Timeout» code. Time of response is about 40 or 50 ms.

I am trying to get NginxUnit server configuration with:
`
curl --unix-socket /var/run/unit/control.sock http://localhost/config/`

and responses like below are being returned:

`curl: (56) Recv failure: Соединение сброшено другой стороной`

or

`curl: (55) Send failure: Обрыв канала`

or empty configuration:

```
{
	"listeners": {},
	"applications": {}
}
```

I have tried to restart the unit process — no effects. I have tried to restart OS — at beginnig it started work well, but then it has been broken again….

I have tried to update NginxUnit from 1.7 to 1.7.1 version. No effects.

I am trying to reconfigurate the unit with:

`curl -X PUT -d @/path/to/json/nginx_unit_config_php73.json --unix-socket /path/to/socket/control.sock http://localhost/config/`

But it answering me with:

`curl: (52) Empty reply from server`

This configuration was earlier loaded into the Unit about moth ago (with modifications):

```
{
    "applications": {
            "my_project_php73": {
            "type": "php 7.3",
            "user": "my_user",
            "group": "my_user",
            "limits": {
            "timeout": 30,
            "requests": 1000
        },
        "processes": 5,
        "root": "//my/project/path/public",
        "script": "index.php",
        "options": {
            "file": "/etc/opt/remi/php73/php.ini",
            "admin": {
                "display_errors": "1",
                "session.save_path": "/my/project/path/mod-tmp",
                "upload_tmp_dir": "/my/project/path/mod-tmp",
                "max_file_uploads": "20",
                "memory_limit": "128M",
                "post_max_size": "100M",
                "upload_max_filesize": "100M"
            }
        }
    },
    "my_project_2_php73": {
        "type": "php 7.3",
        "user": "my_user_2",
        "group": "my_user_2",
        "limits": {
            "timeout": 30,
            "requests": 1000
        },
        "processes": 5,
        "root": "/my/project/path/public",
        "script": "index.php",
        "options": {
            "file": "/etc/opt/remi/php73/php.ini",
            "admin": {
                "display_errors": "0",
                "session.save_path": "/my/project/path/mod-tmp",
                "upload_tmp_dir": "/my/project/path/mod-tmp",
                "max_file_uploads": "20",
                "memory_limit": "128M",
                "post_max_size": "100M",
                "upload_max_filesize": "100M"
            }
        }
    }

    },
    "listeners": {
        "*:8300": {
            "application": "my_project_php73"
        },
        "*:8301": {
            "application": "my_project_2_php73"
        }
    }
}
```

NginxUnit’s log file consists a lot of notes, but I think that these are the most interested:
```

2019/02/12 07:52:07 [error] 20417#20433 *280 send(55, 7F063000E7C8, 7) failed (32: Broken pipe)
2019/02/14 09:20:57.937 [warn] 20419#20419 [unit] port_send(8, 28) failed: Broken pipe (32)
2019/02/14 09:20:57.937 [warn] 20419#20419 [unit] port_send(8, 16) failed: Broken pipe (32)
2019/02/14 09:20:57.937 [alert] 20419#20419 [unit] #794: last message send failed: Broken pipe (32)
2019/02/14 09:21:04 [alert] 6354#6371 *37 writev(-1, 2) failed (9: Bad file descriptor)
2019/02/14 09:21:05.070 [warn] 6355#6355 [unit] failed to send shm to 6354: Broken pipe (32)
2019/02/14 09:21:05.070 [warn] 6355#6355 [unit] port_send(8, 16) failed: Broken pipe (32)
2019/02/14 09:21:05.070 [alert] 6355#6355 [unit] #13: last message send failed: Broken pipe (32)
2019/02/14 09:22:05 [alert] 6399#6415 *41 writev(-1, 2) failed (9: Bad file descriptor)
2019/02/14 09:22:05 [alert] 20413#20413 process 6399 exited on signal 11
2019/02/14 09:22:05 [alert] 20413#20413 sendmsg(14, -1, 2) failed (104: Connection reset by peer)
2019/02/14 09:22:05 [alert] 20413#20413 sendmsg(14, -1, 2) failed (32: Broken pipe)
2019/02/14 09:22:05 [alert] 20413#20413 process 6400 exited on signal 11
```

I don’t know how to resolve this problem and I have temporarily returned to php-fpm.

**Comment by mar0x** [comment]
Hello,

Please provide more information about your installation (OS), try install version with debug logging, reproduce issue and attach unit.log. This gives us information to work with.

Thanks for your report and get ready to turn your installation back to Unit.

**Comment by vsevolod-ch** [comment]
I am sorry, I forgot to write it:
```
CentOS release 6.8 (Final)
PHP 7.3.2
NginxUnit 1.7.1 (after upgrading)
```
And NginxUnit's log file (I have modified paths and names):
[unit.log](https://github.com/nginx/unit/files/2865755/unit.log)

Ok, I am trying to reproduce the issue with debug version the day after tomorrow.

**Comment by mar0x** [comment]
Try to upgrade to 1.7.1-2 to avoid issue #221 .

**Comment by VBart** [comment]
@vsevolod-ch Has the issue been fixed by updating to the new package, or you haven't tried?

**Comment by vsevolod-ch** [comment]
I am sorry, I have not tried yet. I have a lot of work and don't have a time. I hope to update the unit this week. A report will be written.

**Comment by VBart** [comment]
It's ok, no worry. The bad case if you tried the new packaged and it didn't help. In this case we would have another issue to investigate.

**Comment by vsevolod-ch** [comment]
Hello!
I am sorry for the long waiting.
The unit has been updated, and the problem has been solved. I don't know the cause of the issue...
We have returned our php-projects to Unit.
